app/src/main/java/team7/voluntime/Activities/EventRegisterActivity.java

```diff
@@ -70,13 +106,70 @@ protected void onCreate(Bundle savedInstanceState) {
             Log.e(TAG, e.toString());
         }
         eventRegisterEventAddressTV.setText(address);
+        eventRegisterStatusTV.setText("You are not registered for this event");
+
+        reference = database.getReference("Volunteers").child(mUser.getUid());
+        reference.addListenerForSingleValueEvent(new ValueEventListener() {
+            @Override
+            public void onDataChange(@NotNull DataSnapshot dataSnapshot) {
+                if (dataSnapshot.child("Events").child(eventID).getValue() != null) {
+                    eventRegisterButton.setVisibility(View.GONE);
+                    if (dataSnapshot.child("Events").child(eventID).getValue().toString().equals("pending") || dataSnapshot.child("Events").child(eventID).getValue().toString().equals("registered")) {
+                        eventRegisterStatusTV.setText("You have registered for this event");
+                        eventCancelButton.setVisibility(View.VISIBLE);
+                    }
+                    if (dataSnapshot.child("Events").child(eventID).getValue().toString().equals("cancelled")) {
+                        eventRegisterStatusTV.setText("You have cancelled your registration for this event");
+                        eventCancelButton.setVisibility(View.GONE);
```

Comment:
I think you already set this button to 'GONE' in the xml. Do you need to set it again here?

Reply:
It is set to gone then set to visible then set to gone again so you can't cancel again. In latest commit set register button visible again so can reregister